processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -346,6 +353,11 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
         }
     }
 
+    private <T> TraceAndCompletion<T> startTrace(TracingProvider<T> provider, ConsumerRecord<?, ?> record) {
+        T trace = provider.traceFor(record);
+        return new TraceAndCompletion<>(trace, () -> provider.processingCompletion(trace));
```

Comment:
For this point: this call should be surrounded by try-catch too I think https://github.com/line/decaton/pull/69/files#diff-d8bf175344f574cb7bc4d55debac19f5R80